snuba/datasets/outcomes.py

```diff
@@ -51,19 +58,80 @@ def __init__(self):
             ('event_id', Nullable(UUID())),
         ])
 
-        schema = MergeTreeSchema(
-            columns=columns,
+        write_schema = MergeTreeSchema(
+            columns=write_columns,
             # TODO: change to outcomes.raw_local when we add multi DB support
-            local_table_name='outcomes_raw_local',
-            dist_table_name='outcomes_raw_dist',
+            local_table_name=WRITE_SCHEMA_LOCAL_TABLE_NAME,
+            dist_table_name=WRITE_SCHEMA_DIST_TABLE_NAME,
             order_by='(org_id, project_id, timestamp)',
             partition_by='(toMonday(timestamp))',
             settings={
                 'index_granularity': 16384
             })
 
+        read_columns = ColumnSet([
+            ('org_id', UInt(64)),
+            ('project_id', UInt(64)),
+            ('key_id', UInt(64)),
+            ('timestamp', DateTime()),
+            ('outcome', UInt(8)),
+            ('reason', LowCardinality(String())),
+            ('times_seen', UInt(64)),
+        ])
+
+        read_schema = SummingMergeTreeSchema(
+            columns=read_columns,
+            local_table_name=READ_SCHEMA_LOCAL_TABLE_NAME,
+            dist_table_name=READ_SCHEMA_DIST_TABLE_NAME,
+            order_by='(org_id, project_id, key_id, outcome, reason, timestamp)',
+            partition_by='(toMonday(timestamp))',
+            settings={
+                'index_granularity': 256
+            }
+        )
+
+        materialized_view_columns = ColumnSet([
+            ('org_id', UInt(64)),
+            ('project_id', UInt(64)),
+            ('key_id', UInt(64)),
+            ('timestamp', DateTime()),
+            ('outcome', UInt(8)),
+            ('reason', String()),
+            ('times_seen', UInt(64)),
+        ])
+
+        query = """
```

Comment:
I think this approach will have to be revisited, but I guess we need a better separation between logical schema and clockhouse schema first.
What makes me uneasy is the fact that part of the DDL has a nice ColumnSet abstraction and another part is just bare strings so we lack consistency. This problem is magnified for views where the two parts are a more interdependent. 
Let's revisit this later, mind adding a TODO?